components/support/rc_crypto/nss/nss_sys/build.rs

```diff
@@ -56,7 +57,53 @@ fn env_str(name: &str) -> Option<String> {
     env::var(name).ok()
 }
 
+fn env_flag(name: &str) -> bool {
+    match env_str(name).as_ref().map(String::as_ref) {
+        Some("1") => true,
+        Some("0") => false,
+        Some(s) => {
+            println!(
+                "cargo:warning=unknown value for environment var {:?}: {:?}. Ignoring",
+                name, s
+            );
+            false
+        }
+        None => false,
+    }
+}
+const DEFAULT_ANDROID_NDK_API_VERSION: &str = "21";
+
+// Set the CLANG_PATH env variable to point to the right clang for the NDK in question.
+// Note that this basically needs to be done first thing in main.
+fn maybe_setup_ndk_clang_path() {
+    let target_os = env::var("CARGO_CFG_TARGET_OS").ok();
+    if target_os.as_ref().map_or(false, |x| x == "android") {
+        let mut buf = PathBuf::from(env("ANDROID_NDK_ROOT").unwrap());
+        let ndk_api = env_str("ANDROID_NDK_API_VERSION")
+            .unwrap_or(DEFAULT_ANDROID_NDK_API_VERSION.to_owned());
+
+        if ndk_api.is_empty() {
+            println!("cargo:warning=ANDROID_NDK_API_VERSION is unset. Trying unprefixed");
+        }
+        let mut target = env::var("TARGET").unwrap();
+        if target == "armv7-linux-androideabi" {
+            // Yeah, dunno why these are different. The binutils triple is
```

Comment:
If you want to link the reference (https://developer.android.com/ndk/guides/other_build_systems):

(search for "For 32-bit ARM, the compiler is prefixed with armv7a-linux-androideabi")